storage/batch.go

```diff
@@ -84,6 +78,24 @@ func (s *BatchStorage) GetBatchByHash(batchHash common.Hash) (*models.Batch, err
 	return storedBatch.ToModelsBatch(), nil
 }
 
+func (s *BatchStorage) GetPendingBatches() ([]models.Batch, error) {
+	var nilHash *common.Hash
+	storedBatches := make([]stored.Batch, 0)
+	err := s.database.Badger.Find(
+		&storedBatches,
+		bh.Where("Hash").Eq(nilHash).Index("Hash"),
```

Comment:
I think there might be a problem here. At the same time:
* Rollup loop can add a new pending batch
* New block loop can update a batch to add hash after it was mined

As a result both will be updating the same key in Badger and we will have a transaction conflict. What do you think?

Reply:
I don't think it's possible. Rollup loop runs after all available batches are synced, so there shouldn't be update while creating new pending batch.

Reply:
By writing `here` you mean this specific storage function, because it's `find` operation that can't have conflict with `write` operation?

Reply:
>Yup, rollup loop starts after we've synced to the latest block, but then runs in parallel to the new block loop. So you could have a sequence like:
>1. Rollup loop submits batch no. 1.
>2. New block loop enters syncExistingBatch function and update batch no. 1 in storage.
>3. Rollup loop submits batch no. 2 at the same time as step 2 runs.


We discussed that this situation would not happen because we have these execution paths protected by the `stateMutex`.